### PR TITLE
Macroinvertebrate Model

### PR DESCRIPTION
**Pull request by athai**
fixes issue #45 

## Changes in this PR.

- [X] Write tests for macroinvertebrate model
- [x] Implement the model

## Testing this PR.

1. Run ``docker-compose run web bash`` to open up a bash shell
2. cd into streamwebs_frontend/
3. Run ``python manage.py test streamwebs/tests/models`` to run the model tests. Alternatively, you can also run ``python manage.py test streamwebs/tests/*`` to run all the tests!

### Expected Output.

All tests:

```
[root@548a3563b829 streamwebs_frontend]# python manage.py test streamwebs/tests/*
Creating test database for alias 'default'...
...........................Invalid login details: notjohn, badpassword
.Invalid login details: notjohn, johnpassword
.Invalid login details: john, badpassword
.....<ul class="errorlist"><li>username<ul class="errorlist"><li>This field is required.</li></ul></li><li>password<ul class="errorlist"><li>This field is required.</li></ul></li></ul> <ul class="errorlist"><li>school<ul class="errorlist"><li>This field is required.</li></ul></li><li>birthdate<ul class="errorlist"><li>This field is required.</li></ul></li></ul>
...
----------------------------------------------------------------------
Ran 37 tests in 1.205s

OK

```

Only the model tests:

```
 [root@548a3563b829 streamwebs_frontend]# python manage.py test streamwebs/tests/models
Creating test database for alias 'default'...
........................
----------------------------------------------------------------------
Ran 24 tests in 0.409s

OK
Destroying test database for alias 'default'...

```

### CODE REVIEW

File: streamwebs_frontend/streamwebs/tests/models/test_macroinvert.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        self.expected_fields = {
+            'school': models.CharField,
+            'date/time': models.DateTimeField,
+            'weather': models.CharField,
+            'site_name': models.ForeignKey,
+            'water_type': models.CharField,
+            'time_spent': models.PositiveIntegerField,  # in minutes
+            'num_people': models.PositiveIntegerField,
+            'riffle': models.BooleanField,
+            'pool': models.BooleanField,
```
Comment (alxngyn) [inline streamwebs_frontend/streamwebs/tests/models/test_macroinvert.py]: little nitpicking things:

- `date/time` should be something like `date_time`, i feel like that slash will cause headaches in the future
- I couldn't find what you meant by `water_type`on the form

File: streamwebs_frontend/streamwebs/tests/models/test_macroinvert.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  
+            # Somewhat sensitive
+            'clam/mussel': models.PositiveIntegerField,
+            'crane_fly': models.PositiveIntegerField,
+            'crayfish': models.PositiveIntegerField,
+            'damselfly': models.PositiveIntegerField,
+            'dragonfly': models.PositiveIntegerField,
+            'scud': models.PositiveIntegerField,
+            'fishfly': models.PositiveIntegerField,
+            'somewhat_sensitive_total': models.PositiveIntegerField,
```
Comment (alxngyn) [inline streamwebs_frontend/streamwebs/tests/models/test_macroinvert.py]: missing alderfly and mite

**Comment by alxngyn** [comment]
+1

File: streamwebs_frontend/streamwebs/tests/models/test_macroinvert.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+from django.contrib.gis.db import models
+from django.apps import apps
+from itertools import chain
+
+from streamwebs.models import Site
+from streamwebs.models import Macroinvertebrates
+
+
+class SiteTestCase(TestCase):
```
Comment (leian7) [inline streamwebs_frontend/streamwebs/tests/models/test_macroinvert.py]: Maybe rename your class

**Comment by alxngyn** [comment]
++

**Comment by subnomo** [comment]
Looks good. :+1:

**Pull request merged**